flare/framework/algorithm.py

```diff
@@ -83,12 +84,15 @@ class Algorithm(object):
     implement the rest of the network in the Model class.
     """
 
-    def __init__(self, model, hyperparas, gpu_id):
+    def __init__(self, model, gpu_id):
         assert isinstance(model, Model)
         check_duplicate_spec_names(model)
         self.model = model
-        self.hp = hyperparas
-        self.gpu_id = gpu_id
+        if torch.cuda.is_available() and gpu_id >= 0:
+            self.device = "cuda:" + str(gpu_id)
+        else:
+            self.device = "cpu"
+        self.model.to(self.device)
 
     def get_input_specs(self):
         return self.model.get_input_specs()
```

Comment:
better to add comments on the output format of the get_input_specs function. see the comment below